components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -241,8 +241,6 @@ Note that if there is association between a federated user and local user () 'As
             if (ArrayUtils.contains(allowedResourcesForUser, ROOT) || ArrayUtils.contains(allowedResourcesForUser,
                     PERMISSION_ROOT)) {
                 return new ArrayList<>(allScopes);
-            } else if (ArrayUtils.contains(allowedResourcesForUser, ADMIN_PERMISSION_ROOT)) {
-                return new ArrayList<>(getAdminAllowedScopes(allScopes, requestedScopes));
             }
 
             for (Scope scope : allScopes) {
```

Comment:
Can't we change requestedScopes to a set and improve the lookup here?

Reply:
We can remove the if condition by doing it but is it worthy to convert a list to a set and perform set interconnection just to remove an `if` condition?